x-pack/plugins/observability/public/components/rule_kql_filter/kuery_bar.tsx

```diff
@@ -91,7 +83,7 @@ export function MetricsExplorerKueryBar({
           loadSuggestions={curryLoadSuggestions(loadSuggestions)}
           onChange={handleChange}
           onSubmit={onSubmit}
-          placeholder={placeholder || defaultPlaceholder}
+          placeholder={placeholder}
```

Comment:
The placeholder can be empty sometimes, like in (even though it is set as `' '`) -https://github.com/elastic/kibana/blob/933f2a579255b8e6aee536657e9b97bc137bbd56/x-pack/plugins/infra/public/alerting/metric_threshold/components/custom_equation/metric_row_with_count.tsx#L96
Why do we want to make it mandatory?

Reply:
Hmm, I thought we had a default there to ensure we always have a placeholder, if there are valid cases that we don't want a placeholder, then having it optional is fine.

Reply:
The default placeholder was there with `defaultPlaceholder` which contained text `Search for infrastructure data… (e.g. host.name:host-1)`. Since we need different placeholder text in Threshold and APM rules, I moved it to respective rule files, so there is no default place holder now.